crates/sncast/src/helpers/fee.rs

```diff
@@ -83,53 +85,72 @@ impl FeeArgs {
-                        bail!("--max-fee should be greater than or equal to --max-gas amount")
-                    }
-                    (Some(max_fee), None, Some(max_gas_unit_price))
-                        if max_fee < max_gas_unit_price =>
-                    {
-                        bail!("--max-fee should be greater than or equal to --max-gas-unit-price")
-                    }
                     (None, _, _) => FeeSettings::Strk {
-                        max_gas: self.max_gas.map(TryIntoConv::try_into_).transpose()?,
+                        max_gas: self
+                            .max_gas
+                            .map(NonZeroU64::try_from_)
+                            .transpose()
+                            .map_err(anyhow::Error::msg)?,
                         max_gas_unit_price: self
                             .max_gas_unit_price
-                            .map(TryIntoConv::try_into_)
-                            .transpose()?,
-                    },
-                    (Some(max_fee), None, Some(max_gas_unit_price)) => FeeSettings::Strk {
-                        max_gas: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas_unit_price))
-                                .try_into_()?,
-                        ),
-                        max_gas_unit_price: Some(max_gas_unit_price.try_into_()?),
+                            .map(NonZeroU128::try_from_)
+                            .transpose()
+                            .map_err(anyhow::Error::msg)?,
                     },
-                    (Some(max_fee), Some(max_gas), None) => FeeSettings::Strk {
-                        max_gas: Some(max_gas.try_into_()?),
-                        max_gas_unit_price: Some(
-                            max_fee
-                                .floor_div(&NonZeroFelt::from_felt_unchecked(max_gas))
-                                .try_into_()?,
-                        ),
-                    },
-                    (Some(max_fee), None, None) => {
-                        let max_gas_unit_price = provider
-                            .get_block_with_tx_hashes(block_id)
-                            .await?
-                            .l1_gas_price()
-                            .price_in_fri;
+                    (Some(max_fee), None, Some(max_gas_unit_price)) => {
+                        if max_fee < max_gas_unit_price {
+                            bail!(
+                                "--max-fee should be greater than or equal to --max-gas-unit-price"
+                            );
+                        }
+
+                        let max_gas = NonZeroFelt::try_from(Felt::from(max_fee).floor_div(&max_gas_unit_price))
+                        .unwrap_or_else(|_| unreachable!("Calculated max gas must be >= 1 because max_fee >= max_gas_unit_price ensures a positive result"));
```

Comment:
Couldn't some rounding errors still make this <1 in special cases?

Reply:
No, rounding errors can't make this < 1, because we explicitly ensure that `max_fee >= max_gas_unit_price` before this calculation. Since `floor_div` performs exact integer division (rounding down), and both `max_fee` and `max_gas_unit_price` are non-zero, the result of `floor_div` will always be `>=1`.